src/System.Net.Http/src/System/Net/Http/Unix/CurlTransportContext.cs

```diff
@@ -2,33 +2,43 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
-using System;
 using System.Security.Authentication.ExtendedProtection;
+using System.Threading;
 
 namespace System.Net.Http
 {
-    internal class CurlTransportContext : TransportContext
+    internal sealed class CurlTransportContext : TransportContext
     {
-        private CurlChannelBinding _channelBinding = new CurlChannelBinding();
+        private CurlChannelBinding _channelBinding;
 
         internal CurlChannelBinding CurlChannelBinding
         {
             get
             {
-                return _channelBinding;
+                CurlChannelBinding binding = Volatile.Read(ref _channelBinding);
+                if (binding == null)
+                {
+                    var newBinding = new CurlChannelBinding();
+                    binding = Interlocked.CompareExchange(ref _channelBinding, newBinding, null);
```

Comment:
> concerned (though maybe inappropriately?) that GetChannelBinding might be used concurrently with the SSL handshake happening

Normally, the CBT is only available at the conclusion of the SSL handshake. It is a product of the handshake. So, when GetChannelBinding() is able to return something the handshake is over and the secure channel has been established.